crates/core/src/db/datastore/traits.rs

```diff
@@ -243,18 +243,22 @@ pub trait Tx {
     type Tx;
 
     fn begin_tx(&self) -> Self::Tx;
-    fn release_tx(&self, ctx: &ExecutionContext, tx: Self::Tx);
+    fn release_tx<'ctx>(&self, ctx: &'ctx ExecutionContext, tx: Self::Tx) -> RecordTxMetricsOnDrop<'ctx>;
 }
 
 pub trait MutTx {
     type MutTx;
 
     fn begin_mut_tx(&self, isolation_level: IsolationLevel) -> Self::MutTx;
-    fn commit_mut_tx(&self, ctx: &ExecutionContext, tx: Self::MutTx) -> Result<Option<TxData>>;
-    fn rollback_mut_tx(&self, ctx: &ExecutionContext, tx: Self::MutTx);
+    fn commit_mut_tx<'ctx>(
+        &self,
+        ctx: &'ctx ExecutionContext,
+        tx: Self::MutTx,
+    ) -> Result<(TxData, RecordTxMetricsOnDrop<'ctx>)>;
+    fn rollback_mut_tx<'ctx>(&self, ctx: &'ctx ExecutionContext, tx: Self::MutTx) -> RecordTxMetricsOnDrop<'ctx>;
```

Comment:
Without additional context about locking it is quite odd to see this return type, the only thing one can do with is to drop it.

Some prose would certainly appreciated.